src/dataflow-types/src/types.rs

```diff
@@ -299,30 +299,35 @@ impl Arbitrary for SourceInstanceDesc<CollectionMetadata> {
 
     fn arbitrary_with(_: Self::Parameters) -> Self::Strategy {
         (
+            any::<SourceDesc>(),
             any::<SourceInstanceArguments>(),
             any::<CollectionMetadata>(),
         )
-            .prop_map(|(arguments, storage_metadata)| SourceInstanceDesc {
-                description: crate::sources::any_source_desc_stub(),
-                arguments,
-                storage_metadata,
-            })
+            .prop_map(
+                |(description, arguments, storage_metadata)| SourceInstanceDesc {
+                    description,
+                    arguments,
+                    storage_metadata,
+                },
+            )
             .boxed()
     }
 }
 
```

Comment:
Unfortunately not, because of the type parameter `M`. I guess we could add a type bound to `M`, but I'm not familiar enough with how many potential types of metadata we might have and if all of them will have to be `Arbitrary`.